lib/python/picongpu/plugins/plot_mpl/base_visualizer.py

```diff
@@ -0,0 +1,44 @@
+import matplotlib.pyplot as plt
+
+
+class Visualizer(object):
+    def __init__(self, run_directory):
+
+        if run_directory is None:
+            raise ValueError('The run_directory parameter can not be None!')
+
+        self.plt_obj = None
+        self.data_reader = self._create_data_reader(run_directory)
+        self.data = None
+
+    def _create_data_reader(self, run_directory):
+        raise NotImplementedError
+
+    def _create_plt_obj(self, ax):
```

Comment:
what is the matplotlib baseclass this "plot_obj" refers to?
Do you mean an [`matplotlib.artist.Artist`](https://matplotlib.org/api/artist_api.html#matplotlib.artist.Artist)?

* https://matplotlib.org/users/artists.html

(I checked `plt.plot` which returns a list of Line2D which is a child class of `matplotlib.artist.Artist`)

Reply:
honestly I have not looked at whether all those matplotlib plotting objects have a common baseclass...What I meant with `self.plt_obj` was any object coming from matplotlib for which you could update data...

Reply:
`plt.imshow` also returns an `matplotlib.artist.Artist`, so let use be explicit here since we are mpl only.

Reply:
But its always some function of the axes that we call, e.g. `ax.imshow()` or `ax.semilogy()` and they all seem to return different kinds of objects